pkg/kncloudevents/message_receiver.go

```diff
@@ -37,6 +37,8 @@ type HTTPMessageReceiver struct {
 
 	server   *http.Server
 	listener net.Listener
+
+	checker http.HandlerFunc
 }
 
 func NewHTTPMessageReceiver(port int) *HTTPMessageReceiver {
```

Comment:
How about using option style here to be more flexible?

`func NewHTTPMessageReceiver(port int, opts... XXOptions) *HTTPMessageReceiver `

Reply:
Done.